utils/swift_build_support/swift_build_support/products/product.py

```diff
@@ -366,6 +380,31 @@ def generate_linux_toolchain_file(self, platform, arch):
 
         return toolchain_file
 
+    def generate_toolchain_file(self, host_target):
```

Comment:
Given the usage in the cmark product, could it make sense to explicitly state that the function targets darwin and linux only? e.g.

```suggestion
    def generate_toolchain_file_for_darwin_and_linux(self, host_target):
```